src/core/core/library_utils.cpp

```diff
@@ -71,7 +71,7 @@ namespace dsn {
             }            
 # elif defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
             module_name = "lib" + module_name + ".so";
-            auto hmod = dlopen(module_name.c_str(), RTLD_LAZY);
+            auto hmod = dlopen(module_name.c_str(), RTLD_LAZY|RTLD_GLOBAL);
```

Comment:
_Hi zuoyan, thanks for pointing it out. My bad, I should explain the context more clearly in the commit history._
The reason is that when I was trying to deploy rDSN.monitor on Linux, I met **ImportError** such as:
```python
undefined symbol: _Py_ZeroStruct, PyFloat_Type
```
And the load link is like:
**simple_kv(executable)** --dynamically-load-> **libdev.python_helper.so** --load-> **python interpreter**
To solve this issue, I thought using RTLD_GLOBAL would make symbols available for libraries loaded afterwards.

**Yes**, your concern is right. Another alternative is to link Python lib with executables, to which I hold my opinion. I would consult Zhenyu tomorrow to see what he would suggest.